scheduler/src/cook/mesos/api.clj

```diff
@@ -1829,158 +1829,158 @@
-        {:produces ["application/json"],
-         :get {:summary "Read reasons why a job isn't being scheduled."
-               :parameters {:query-params UnscheduledJobParams}
-               :handler (read-unscheduled-handler conn is-authorized-fn)
-               :responses {200 {:schema UnscheduledJobResponse
-                                :description "Reasons the job isn't being scheduled."}
-                           400 {:description "Invalid request format."}
-                           404 {:description "The UUID doesn't correspond to a job."}}}})))
-    (ANY "/queue" []
-         (waiting-jobs mesos-pending-jobs-fn is-authorized-fn mesos-leadership-atom leader-selector))
-    (ANY "/running" []
-         (running-jobs conn is-authorized-fn))
-    (ANY "/list" []
-         (list-resource (db conn) framework-id is-authorized-fn)))
-   (format-params/wrap-restful-params {:formats [:json-kw]
-                                       :handle-error c-mw/handle-req-error})
-   (streaming-json-middleware)))
+              :handler (groups-action-handler conn task-constraints is-authorized-fn)}}))
+
+        (c-api/context
+          "/failure_reasons" []
+          (c-api/resource
+            {:get {:summary "Returns a description of all possible task failure reasons"
+                   :responses {200 {:schema FailureReasonsResponse
+                                    :description "The failure reasons were returned."}}
+                   :handler (failure-reasons-handler conn is-authorized-fn)}}))
+
+        (c-api/context
+          "/settings" []
+          (c-api/resource
+            {:get {:summary "Returns the settings that cook is configured with"
+                   :responses {200 {:description "The settings were returned."}}
+                   :handler (settings-handler settings)}}))
+
+        (c-api/context
+          "/unscheduled_jobs" []
+          (c-api/resource
+            {:produces ["application/json"],
+             :get {:summary "Read reasons why a job isn't being scheduled."
+                   :parameters {:query-params UnscheduledJobParams}
+                   :handler (read-unscheduled-handler conn is-authorized-fn)
+                   :responses {200 {:schema UnscheduledJobResponse
+                                    :description "Reasons the job isn't being scheduled."}
+                               400 {:description "Invalid request format."}
+                               404 {:description "The UUID doesn't correspond to a job."}}}})))
+      (ANY "/queue" []
+        (waiting-jobs mesos-pending-jobs-fn is-authorized-fn mesos-leadership-atom leader-selector))
+      (ANY "/running" []
+        (running-jobs conn is-authorized-fn))
+      (ANY "/list" []
+        (list-resource (db conn) framework-id is-authorized-fn)))
+    (format-params/wrap-restful-params {:formats [:json-kw]
+                                        :handle-error c-mw/handle-req-error})
+    (streaming-json-middleware)))
```

Comment:
Nitpick: It would be nice if we could prevent improperly-intended code from being checked in. We should all settle on a standard formatting that we're happy with, then make a [cljfmt](https://github.com/weavejester/cljfmt) configuration that we can run locally and in Travis.